server/src/main/java/com/objectcomputing/checkins/services/role/role_permissions/RolePermissionServicesImpl.java

```diff
@@ -53,14 +53,33 @@ public List<RolePermissionsResponseDTO> findAll() {
     }
 
     @Override
-    public RolePermission save(UUID roleId, UUID permissionId) {
-        rolePermissionRepository.saveByIds(roleId.toString(), permissionId.toString());
-        RolePermission saved = rolePermissionRepository.findByIds(roleId.toString(), permissionId.toString()).get(0);
+    public RolePermission save(UUID roleId, Permission permissionId) {
+        rolePermissionRepository.saveByIds(roleId.toString(), permissionId);
+        RolePermission saved = rolePermissionRepository.findByIds(roleId.toString(), permissionId).get(0);
         return saved;
     }
 
     @Override
-    public void delete(UUID roleId, UUID permissionId) {
-        rolePermissionRepository.deleteByIds(roleId.toString(), permissionId.toString());
+    public void delete(UUID roleId, Permission permissionId) {
+        rolePermissionRepository.deleteByIds(roleId.toString(), permissionId);
+    }
+
+    @Override
+    public List<RolePermission> findByRoleId(UUID roleId) {
+        return rolePermissionRepository.findByRoleId(roleId);
+    }
+
+    @Override
+    public List<Permission> findUserPermissions(@NotBlank UUID id) {
+
+        Set<Role> memberRoles = roleServices.findUserRoles(id);
+
+        return memberRoles.stream().map(role ->
+                findByRoleId(role.getId())
+                    .stream()
+                    .map(RolePermission::getPermission)
+                    .collect(Collectors.toList()))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
```

Comment:
Right now this method doesn't have any tests associated with it. It may be worth adding a test or two in `CurrentUserControllerTest` to assign a user a role or multiple roles, and ensure the response has the correct permissions.